roles/kubernetes/master/templates/kubeadm-config.yaml.j2

```diff
@@ -83,3 +83,7 @@ apiServerCertSANs:
 {% endfor %}
 certificatesDir: {{ kube_config_dir }}/ssl
 unifiedControlPlaneImage: "{{ hyperkube_image_repo }}:{{ hyperkube_image_tag }}"
+{% if kube_override_hostname|default('') %}
+nodeName: {{ inventory_hostname }}
```

Comment:
It has to be:

nodeName: {{ kube_override_hostname }}

Reply:
whoops. good eye.